### PR TITLE
Add eli5_category dataset

### PR DESCRIPTION
**Pull request by jingshenSN2**
This pull request adds a categorized Long-form question answering dataset `ELI5_Category`. It's a new variant of the [ELI5](https://huggingface.co/datasets/eli5) dataset that uses the Reddit tags to alleviate the training/validation overlapping in the origin ELI5 dataset.

A [report](https://celeritasml.netlify.app/posts/2021-12-01-eli5c/)(Section 2) on this dataset.

### CODE REVIEW

File: datasets/eli5_category/README.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+licenses:
+- unknown
+multilinguality:
+- monolingual
+paperswithcode_id: eli5-category
+pretty_name: ELI5-Category
+size_categories:
+- 100K<n<1M
+source_datasets:
+- original
```
Comment (lhoestq) [inline datasets/eli5_category/README.md:17]:
```suggestion
- extended|eli5
```

File: datasets/eli5_category/README.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+- found
+language_creators:
+- found
+languages:
+- en
+licenses:
+- unknown
+multilinguality:
+- monolingual
+paperswithcode_id: eli5-category
```
Comment (lhoestq) [inline datasets/eli5_category/README.md:12]: I think it doesn't exist on PWC:
https://paperswithcode.com/dataset/eli5-category returns "Not Found"

```suggestion
paperswithcode_id: null
```

File: datasets/eli5_category/eli5_category.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+            datasets.SplitGenerator(
+                name=datasets.Split("validation1"),
+                gen_kwargs={"filepath": downloaded_files["val1"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split("validation2"),
+                gen_kwargs={"filepath": downloaded_files["val2"]},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split("test"),
```
Comment (lhoestq) [inline datasets/eli5_category/eli5_category.py:106]: nit:
```suggestion
                name=datasets.Split.TEST,
```

**Review by lhoestq — APPROVED** [review]
Thanks a lot for adding this dataset ! Good job with the dataset card and the dataset scripts - they're really good :)

**Comment by jingshenSN2** [comment]
> Thanks a lot for adding this dataset ! Good job with the dataset card and the dataset scripts - they're really good :)
> 
> I just added minor changes

Thanks for fixing typos!
